File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/LocationRegionMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.LocationRegionDTO;
import com.tu.votingapp.entities.LocationRegionEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = {LocationMapper.class})
public interface LocationRegionMapper {
    LocationRegionDTO toDto(LocationRegionEntity entity);
    LocationRegionEntity toEntity(LocationRegionDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/LocationRegionMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.LocationRegionResponseDTO
import com.example.android.entity.LocationEntity
import com.example.android.entity.LocationRegionEntity
import com.example.android.entity.MunicipalityEntity
import com.example.android.entity.RegionEntity
import com.example.android.utils.mappers.LocationMapper.toDto

object LocationRegionMapper {

    fun LocationRegionResponseDTO.toEntity(): LocationRegionEntity = LocationRegionEntity(
        id = this.id,
        name = this.name,
        locationId = this.location.id
    )

    fun LocationRegionEntity.toDto(location: LocationEntity, municipality: MunicipalityEntity, region: RegionEntity): LocationRegionResponseDTO = LocationRegionResponseDTO(
        id = this.id!!,
        name = this.name,
        location = location.toDto(municipality, region)
    )
}